src/Microsoft.AspNet.Server.Kestrel/Http/Frame.Generated.cs

```diff
@@ -0,0 +1,297 @@
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Server.Kestrel.Http 
+{
+    public partial class Frame
+    {
+        
+        private object _currentIHttpRequestFeature;
+        private object _currentIHttpResponseFeature;
+        private object _currentIHttpRequestIdentifierFeature;
+        private object _currentIHttpSendFileFeature;
+        private object _currentIServiceProvidersFeature;
+        private object _currentIHttpAuthenticationFeature;
+        private object _currentIHttpRequestLifetimeFeature;
+        private object _currentIQueryFeature;
+        private object _currentIFormFeature;
+        private object _currentIResponseCookiesFeature;
+        private object _currentIItemsFeature;
+        private object _currentIHttpConnectionFeature;
+        private object _currentITlsConnectionFeature;
+        private object _currentIHttpUpgradeFeature;
+        private object _currentIHttpWebSocketFeature;
+        private object _currentISessionFeature;
+
+        private void FastReset()
+        {
+            _currentIHttpRequestFeature = this as global::Microsoft.AspNet.Http.Features.IHttpRequestFeature;
+            _currentIHttpResponseFeature = this as global::Microsoft.AspNet.Http.Features.IHttpResponseFeature;
+            _currentIHttpRequestIdentifierFeature = this as global::Microsoft.AspNet.Http.Features.IHttpRequestIdentifierFeature;
+            _currentIHttpSendFileFeature = this as global::Microsoft.AspNet.Http.Features.IHttpSendFileFeature;
+            _currentIServiceProvidersFeature = this as global::Microsoft.AspNet.Http.Features.Internal.IServiceProvidersFeature;
+            _currentIHttpAuthenticationFeature = this as global::Microsoft.AspNet.Http.Features.Authentication.IHttpAuthenticationFeature;
+            _currentIHttpRequestLifetimeFeature = this as global::Microsoft.AspNet.Http.Features.IHttpRequestLifetimeFeature;
+            _currentIQueryFeature = this as global::Microsoft.AspNet.Http.Features.Internal.IQueryFeature;
+            _currentIFormFeature = this as global::Microsoft.AspNet.Http.Features.Internal.IFormFeature;
+            _currentIResponseCookiesFeature = this as global::Microsoft.AspNet.Http.Features.Internal.IResponseCookiesFeature;
+            _currentIItemsFeature = this as global::Microsoft.AspNet.Http.Features.Internal.IItemsFeature;
+            _currentIHttpConnectionFeature = this as global::Microsoft.AspNet.Http.Features.IHttpConnectionFeature;
+            _currentITlsConnectionFeature = this as global::Microsoft.AspNet.Http.Features.ITlsConnectionFeature;
+            _currentIHttpUpgradeFeature = this as global::Microsoft.AspNet.Http.Features.IHttpUpgradeFeature;
+            _currentIHttpWebSocketFeature = this as global::Microsoft.AspNet.Http.Features.IHttpWebSocketFeature;
+            _currentISessionFeature = this as global::Microsoft.AspNet.Http.Features.ISessionFeature;
```

Comment:
Yeah, I was thinking that also. The `as` cast is just so the generated code remains correct when features are added or removed from the frame.

That said - it's probably better to hard-code those assignments at code-generate time and put a comment on the Frame partial that implements interfaces telling the developer to change the related file.